src-java/blue-green/src/main/java/org/openkilda/bluegreen/ZkClient.java

```diff
@@ -141,18 +167,31 @@ protected void ensureZNode(byte[] value, String... path) throws KeeperException,
         }
     }
 
-    void validateNodes() throws KeeperException, InterruptedException, IOException {
+    @VisibleForTesting
+    void validateZNodes() throws KeeperException, InterruptedException, IOException {
         ensureZNode(serviceName);
         ensureZNode(serviceName, id);
+        validateNodes();
+        nodesValidated = true;
```

Comment:
You are making an assumption that nothing can happen with nodes while there where no connection to zk... as for me it is too optimistic.

Reply:
not exactly, if zk is reconnected the flag is reseted to false